src/panels/config/automation/action/types/ha-automation-action-event.ts

```diff
@@ -73,8 +74,11 @@ export class HaEventAction extends LitElement implements ActionElement {
     handleChangeEvent(this, ev);
   }
 
-  private _valueChanged(ev: CustomEvent): void {
-    handleChangeEvent(this, ev);
+  private _eventChanged(ev: CustomEvent): void {
+    ev.stopPropagation();
+    fireEvent(this, "value-changed", {
+      value: { ...this.action, entity_id: ev.detail.value },
```

Comment:
I assume this should be event?
```suggestion
      value: { ...this.action, event: ev.detail.value },
```